task.py

```diff
@@ -275,16 +276,40 @@ def get_namespace(self) -> str:
     def get_duration(self) -> int:
         return self.ts.cfg_descr.get_tft().duration
 
+    def _fetch_default_resource_name(self) -> Union[str, None]:
+        k8sClient = K8sClient()
+        data = k8sClient.oc_get("network-attachment-definition")
+
+        if isinstance(data, dict):
+            resource_name = data["items"][0]["metadata"]["annotations"].get(
+                "k8s.v1.cni.cncf.io/resourceName"
+            )
+
+        if isinstance(resource_name, str):
+            return resource_name
+        else:
+            return None
+
     def get_template_args(self) -> dict[str, str | list[str]]:
+        extra: dict[str, Any] = {}
+        if isinstance(self.ts.connection.secondary_network_nad, str):
+            common.dict_add_optional(
+                extra, "secondary_network_nad", self.ts.connection.secondary_network_nad
+            )
+        if isinstance(self.ts.connection.resource_name, str):
+            common.dict_add_optional(
```

Comment:
The `if isinstance` check is not necessary.

The point of `common.dict_add_optional()` is to take an `Optional[T]` value, and add the entry to the dict if-and-only-if the value is not None.